plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/delete/DeletionVectors.java

```diff
@@ -67,6 +74,47 @@ public static Roaring64NavigableMap readDeletionVectors(TrinoFileSystem fileSyst
         throw new IllegalArgumentException("Unexpected storage type: " + deletionVector.storageType());
     }
 
+    public static DeletionVectorEntry writeDeletionVectors(
+            TrinoFileSystem fileSystem,
+            Location location,
+            Roaring64Bitmap deletedRows)
+            throws IOException
+    {
+        UUID uuid = randomUUID();
+        String deletionVectorFilename = "deletion_vector_" + uuid + ".bin";
+        String pathOrInlineDv = encodeUuid(uuid);
+        RoaringBitmapArray bitmapArray = new RoaringBitmapArray(deletedRows);
+        int sizeInBytes = MAGIC_NUMBER_BYTE_SIZE + BIT_MAP_COUNT_BYTE_SIZE + BIT_MAP_KEY_BYTE_SIZE + bitmapArray.serializedSizeInBytes();
+        long cardinality = deletedRows.getLongCardinality();
+
+        checkArgument(sizeInBytes > 0, "sizeInBytes must be positive: %s", sizeInBytes);
+        checkArgument(cardinality > 0, "cardinality must be positive: %s", cardinality);
+
+        OptionalInt offset;
+        byte[] data = serializeAsByteArray(bitmapArray.bitmaps(), sizeInBytes);
+        try (DataOutputStream output = new DataOutputStream(fileSystem.newOutputFile(location.appendPath(deletionVectorFilename)).create())) {
+            output.writeByte(FORMAT_VERSION_V1);
+            offset = OptionalInt.of(output.size());
+            output.writeInt(sizeInBytes);
+            output.write(data);
+            output.writeInt(calculateChecksum(data));
+        }
+
+        return new DeletionVectorEntry(UUID_MARKER, pathOrInlineDv, offset, sizeInBytes, cardinality);
+    }
+
+    private static byte[] serializeAsByteArray(RoaringBitmap[] bitmaps, int sizeInBytes)
+    {
+        ByteBuffer buffer = ByteBuffer.allocate(sizeInBytes).order(LITTLE_ENDIAN);
+        buffer.putInt(PORTABLE_ROARING_BITMAP_MAGIC_NUMBER);
+        buffer.putLong(bitmaps.length);
+        for (int i = 0; i < bitmaps.length; i++) {
+            buffer.putInt(i); // Bitmap index
+            bitmaps[i].serialize(buffer);
```

Comment:
It's usually good idea to call `runOptimize()` on roaring bitmaps before writing them